tests/backup_helper.go

```diff
@@ -10471,3 +10493,341 @@ func IsPxInstalled() bool {
+func UnShareCluster(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) (*api.UnShareClusterResponse, error) {
+	backupDriver := Inst().Backup
+	unshareClusterRequest := &api.UnShareClusterRequest{
+		OrgId: BackupOrgID,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  clusterUid,
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+	}
+	resp, err := backupDriver.UnShareCluster(ctx, unshareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+	return resp, nil
+}
+
+// UnShareClusterWithValidation unshares a cluster with users and groups and validates the unshare.
+func UnShareClusterWithValidation(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) (*api.UnShareClusterResponse, error) {
+	backupDriver := Inst().Backup
+	unshareClusterRequest := &api.UnShareClusterRequest{
+		OrgId: BackupOrgID,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  clusterUid,
+		},
+		// userid of the user(s) to share the cluster with
+		Users: users,
+		// group(s) to share the cluster with
+		Groups: groups,
+	}
+	resp, err := backupDriver.UnShareCluster(ctx, unshareClusterRequest)
+	if err != nil {
+		return resp, err
+	}
+
+	err = ValidateUnShareCluster(ctx, clusterName, clusterUid, users, groups)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// ValidateShareCluster validates that a cluster is shared with users and groups by validating collaborators list and inspecting cluster from user.
+func ValidateShareCluster(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) error {
+	backupDriver := Inst().Backup
+	userIds := []string{}
+	groupIds := []string{}
+	req := &api.ClusterInspectRequest{
+		Name:  clusterName,
+		OrgId: BackupOrgID,
+		Uid:   clusterUid,
+	}
+
+	// Inspect cluster to retrieve collaborators
+	clusterObject, err := backupDriver.InspectCluster(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	// Extract collaborator user and group IDs
+	clusterCollaborators := clusterObject.GetCluster().Ownership
+	for _, user := range clusterCollaborators.Collaborators {
+		userIds = append(userIds, user.Id)
+	}
+	for _, group := range clusterCollaborators.Groups {
+		groupIds = append(groupIds, group.Id)
+	}
+
+	// Validate that users and groups are present in the collaborators list
+	for _, user := range users {
+		if !IsPresent(userIds, user) {
+			return fmt.Errorf("User with id [%s] is not present in the list of collaborators for cluster [%s]", user, clusterName)
+		}
+	}
+	for _, group := range groups {
+		if !IsPresent(groupIds, group) {
+			return fmt.Errorf("Group with id [%s] is not present in the list of collaborators for cluster [%s]", group, clusterName)
+		}
+	}
+
+	// Validate that the entire list matches
+	if !AreStringSlicesEqual(userIds, users) {
```

Comment:
Is this required?

Reply:
This will ensure that the fetched list from collaborators doesn't contain any extra entries.